### PR TITLE
[NOD-758] Add confirmations to blocks

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

**Review by dereeno — DISMISSED** [review]

File: kasparovd/controllers/block.go [hunk 1]
```diff
@@ -31,7 +32,14 @@ func GetBlockByHashHandler(blockHash string) (interface{}, error) {
 		return nil, httpserverutils.NewHandlerError(http.StatusNotFound, errors.New("no block with the given block hash was found"))
 	}
 
-	return apimodels.ConvertBlockModelToBlockResponse(block), nil
+	selectedTipBlueScore, err := dbaccess.SelectedTipBlueScore(dbaccess.NoTx())
+	if err != nil {
+		return nil, err
+	}
+
+	blockRes := apimodels.ConvertBlockModelToBlockResponse(block)
```
Comment (svarogg) [inline kasparovd/controllers/block.go]: Pass selectedTipBlueScore and than assign confirmations inside

File: kasparovd/controllers/block.go [hunk 2]
```diff
@@ -51,9 +59,15 @@ func GetBlocksHandler(orderString string, skip uint64, limit uint64) (interface{
...
 
+	selectedTipBlueScore, err := dbaccess.SelectedTipBlueScore(dbaccess.NoTx())
+	if err != nil {
+		return nil, err
+	}
+
 	blockResponses := make([]*apimodels.BlockResponse, len(blocks))
 	for i, block := range blocks {
 		blockResponses[i] = apimodels.ConvertBlockModelToBlockResponse(block)
+		blockResponses[i].Confirmations = pointers.Uint64(confirmations(blockResponses[i].AcceptingBlockBlueScore, selectedTipBlueScore))
```
Comment (svarogg) [inline kasparovd/controllers/block.go]: Here too

**Review by svarogg — CHANGES_REQUESTED** [review]

**Review by svarogg — DISMISSED** [review]

**Review by dereeno — DISMISSED** [review]

File: apimodels/conversions.go [hunk 3]
```diff
@@ -70,8 +82,47 @@ func ConvertBlockModelToBlockResponse(block *dbmodels.Block) *BlockResponse {
...
 		blockRes.ParentBlockHashes[i] = parent.BlockHash
 	}
 	return blockRes
 }
+
+// ConvertTransactionOutputModelToTransactionOutputResponse converts a transaction output
+// database object into a TransactionOutputResponse
+func ConvertTransactionOutputModelToTransactionOutputResponse(transactionOutput *dbmodels.TransactionOutput,
+	selectedTipBlueScore uint64, activeNetParams *dagconfig.Params, isSpent bool) (*TransactionOutputResponse, error) {
+	subnetworkID := &subnetworkid.SubnetworkID{}
```
Comment (svarogg) [inline apimodels/conversions.go:97]: Please add empty line after multi-line function declaration

File: apimodels/conversions.go [hunk 4]
```diff
@@ -68,9 +80,49 @@ func ConvertBlockModelToBlockResponse(block *dbmodels.Block) *BlockResponse {
...
+		Value:                   transactionOutput.Value,
+		ScriptPubKey:            hex.EncodeToString(transactionOutput.ScriptPubKey),
+		AcceptingBlockHash:      acceptingBlockHash,
+		AcceptingBlockBlueScore: acceptingBlockBlueScore,
+		Index:                   transactionOutput.Index,
+		IsCoinbase:              &isCoinbase,
+		Confirmations:           &utxoConfirmations,
+	}
+
+	isSpendable := false
```
Comment (svarogg) [inline apimodels/conversions.go]: Better to first calculate `isSpendable`, and only than instantiate `txOutRes`, with isSpendable already known, just as we do with `isCoinbase` and `utxoConfirmation`

**Review by svarogg — CHANGES_REQUESTED** [review]

**Review by svarogg — DISMISSED** [review]

File: apimodels/conversions.go [hunk 5]
```diff
@@ -2,12 +2,23 @@ package apimodels
...
 	"encoding/hex"
+	"github.com/kaspanet/kaspad/dagconfig"
+	"github.com/kaspanet/kaspad/util/pointers"
+	"github.com/kaspanet/kaspad/util/subnetworkid"
+	"github.com/pkg/errors"
 
 	"github.com/kaspanet/kasparov/dbmodels"
 )
 
+func confirmations(acceptingBlueScore *uint64, selectedTipBlueScore uint64) uint64 {
```
Comment (dereeno) [inline apimodels/conversions.go]: please rename to acceptingBlockBlueScore to be consistent with the other places we use it

**Review by dereeno — CHANGES_REQUESTED** [review]

**Review by dereeno — DISMISSED** [review]

**Review by svarogg — APPROVED** [review]

**Review by dereeno — APPROVED** [review]

**Pull request merged**